### PR TITLE
[PROTOCOL] Clarify checkpoint names

### LINKED ISSUE #2842: [PROTOCOL RFC] Clarify file names of multi-part checkpoint files
### Description of the protocol change

The naming convention for multi-part checkpoint files is ambiguous:

> A multi-part checkpoint for version n consists of p "part" files (p > 1), where part o of p is named n.checkpoint.o.p.parquet.

It's not specified that `o` and `p` are zero padded numbers of length 10. It would also be good to state explicitly that `n` is a zero padded number of length 20 for multi-part checkpoints and the other types of checkpoints. 

We should clarify this in the protocol.

### Willingness to contribute

The Delta Lake Community encourages protocol innovations. Would you or another member of your organization be willing to contribute this feature to the Delta Lake code base?

- [x] Yes. I can contribute.
- [ ] Yes. I would be willing to contribute with guidance from the Delta Lake community.
- [ ] No. I cannot contribute at this time.

### PR DESCRIPTION
**Pull request by jkylling**
#### Which Delta project/connector is this regarding?

- [ ] Spark
- [ ] Standalone
- [ ] Flink
- [ ] Kernel
- [x] Other (Protocol)

## Description

Resolves #2842

## Does this PR introduce _any_ user-facing changes?

No

### CODE REVIEW

**Comment by jkylling** [comment]
@dennyglee could this be assigned a reviewer? :pray:

File: PROTOCOL.md [hunk 1]
```diff
@@ -220,6 +220,7 @@ _sidecars/7d17ac10-5cc3-401b-bd1a-9c82dd2ea032.parquet
 ```
 
 2. A [classic checkpoint](#classic-checkpoint) for version `n` of the table consists of a file named `n.checkpoint.parquet`.
+Here `n` must be zero padded to have length 20.
```
Comment (tdas) [inline PROTOCOL.md:223]: can just put this in the previous line.

File: PROTOCOL.md [hunk 2]
```diff
@@ -229,7 +230,8 @@ For example:
 
 
 3. A [multi-part checkpoint](#multi-part-checkpoint) for version `n` consists of `p` "part" files (`p > 1`), where
-part `o` of `p` is named `n.checkpoint.o.p.parquet`. These are always [V1 checkpoints](#v1-spec).
+part `o` of `p` is named `n.checkpoint.o.p.parquet`. Here `n` must be zero padded to have length 20, while `o` and `p` must be zero padded to have length 10.
+These are always [V1 checkpoints](#v1-spec).
```
Comment (tdas) [inline PROTOCOL.md:234]: can put this in previous line.

**Review by tdas — APPROVED** [review]
This is a very good clarification. @ryan-johnson-databricks can you confirm the correctness of this?

**Review by prakharjain09 — APPROVED** [review]

**Comment by prakharjain09** [comment]
> @ryan-johnson-databricks can you confirm the correctness of this?

Yes this is correct. The 20 character zero padding is needed to make sure lexiographical listing puts them in the right place. It is mentioned at couple of places in the spec already.

Thanks for fixing the rest.

**Pull request merged**